Source/ACE/Network/GameAction/Actions/GameActionFinishBarber.cs

```diff
@@ -13,7 +7,9 @@ public static class GameActionFinishBarber
         [GameAction(GameActionType.FinishBarber)]
         public static void Handle(ClientMessage message, Session session)
         {
-            session.Player.HandleActionFinishBarber(message);
+            ActionChain chain = new ActionChain();
+            chain.AddAction(session.Player, () => session.Player.HandleActionFinishBarber(message));
```

Comment:
I would prefer that the action chain is within HandleActionFInishBarber (HandleAction indicates that the function is safe to call from outside of the player's action context).  I wont block the PR over it.